docs/spec/data-plane.md

```diff
@@ -0,0 +1,76 @@
+# Knative Eventing Data Plane Contracts
+
+## Introduction
+
+Developers using Knative Eventing need to know what is supported for delivery to user provided components that receive events. Knative Eventing defines contract for data plane components and we have listed them here.
+
+## Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC2119.
+
+## Data plane contract for Sinks
+
+A **Sink** is an [_addressable_](./interfaces.md#addressable) resource that takes responsibility for the event. A Sink could be a consumer of events, or middleware. A Sink MUST be able to receive CloudEvents over HTTP and HTTPS.
+
+A **Sink** MAY be [_callable_](./interfaces.md#callable) resource that represents an Addressable endpoint which receives an event as input and optionally returns an event to forward downstream
+
+Almost every component in Knative Eventing may be a Sink providing composability.
+
+Every Sink MUST support HTTP Protocol Binding for CloudEvents [version 1.0](https://github.com/cloudevents/spec/blob/v1.0/http-protocol-binding.md) and [version 0.3](https://github.com/cloudevents/spec/blob/v0.3/http-transport-binding.md) with restrictions and extensions specified below.
+
+### HTTP Support
```

Comment:
Something to consider - CloudEvents HTTP Binary encoding makes it trial to support authorization policies based on CE attributes as they are present in the HTTP headers.

Supporting structured encoding makes it so auth decisions could have to parse an up to 10MB JSON payload (per CE spec on size).

Reply:
The question being - should we consider only supporting HTTP binary encoding in eventing infrastructure (both both sending and receiving)

Reply:
I see your point. 
On the other hand you could have a source that retrieves a lot of events in structured encoding but with a size <1KB. There might be good reasons to use structured encoding. The source might subscribe to some MQTT 3.1.1 topics, where only structured encoding is possible.  So you would require this source to transcode all events.